vars/parseStageInfo.groovy

```diff
@@ -52,6 +52,14 @@ def call(Map config = [:]) {
       }
     }
 
+    if (result['target'].startsWith('centos')) {
+      result['java_pkg'] = 'java-1.8.0-openjdk'
+    } else if (result['target'].startsWith('ubuntu')) {
+      result['target'] = 'openjdk-8-jdk'
+    } else if (result['target'].startsWith('leap')) {
+      result['target'] = 'java-1_8_0-openjdk'
+    }
```

Comment:
It would be nice to be obvious here if a match couldn't be found rather than somebody having to back track it.
```suggestion
    } else {
        error("Don't know which java package to install for " + result['target'])
    }
```

Reply:
If we add another distro to our support, we are likely going to have to make many changes in this file and others.

By leaving the key unspecified with out a match, it is up to the routine that consumes that information to generate the diagnostic if it is missing.  We are also now pre-installing the Java package on the CI nodes.

Reply:
But then we are going to be repeating a lot of code in all of the callers to validate the info was parsed correctly.  It seems more efficient and the right place to flag missing information where it's being parsed/detected.

Reply:
I have put that in for the next push.

Someone adding a new target distribution for a stage that did not need Java might be surprised when this error shows up.

Reply:
I can see that perspective, indeed.  But I think porting a new distro should be complete anyway, so that future expansion doesn't need to deal with a previously incomplete distro port.

But saying that makes me think somehow getting all of these various distro variations across our many files and functions into a single location and then referenced from there would be a good enhancement one day.  A single place to go put all of the bits needed to teach our CI about a new distro.